src/pure/target.ml

```diff
@@ -670,7 +652,7 @@ type state = [
       | history -> Some { history }
 
     let merge one two =
-      let compare a b = ~- (Float.compare a.time b.time) in
+      let compare a b = ~- (Int.compare a.depth b.depth) in
```

Comment:
`~-` is an alias of `-` which OCaml understands as unary-prefix operator without guessing (`-` can be both unary and binary — the default, I think — so in doubt I use `~-`).